mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BucketNameGenerator.java

```diff
@@ -0,0 +1,41 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.blob;
+
+import java.time.Clock;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoField;
+
+import org.apache.james.blob.api.BucketName;
+
+public class BucketNameGenerator {
+    private final Clock clock;
+
+    public BucketNameGenerator(Clock clock) {
+        this.clock = clock;
+    }
+
+    public BucketName currentBucket() {
+        ZonedDateTime now = ZonedDateTime.now(clock);
+        int month = now.getMonth().get(ChronoField.MONTH_OF_YEAR);
+        int year = now.getYear();
+        return BucketName.of(String.format("deletedMessages-%d-%d-01", month, year));
```

Comment:
I would put the year before the month, it's usually more practical

Reply:
So would I (and did I)